src/domain/usecases/SendNotificationDataValuesUseCase.ts

```diff
@@ -0,0 +1,347 @@
+import _ from "lodash";
+import jsonfile from "jsonfile";
+import log from "utils/log";
+import { DataValuesRepository } from "domain/repositories/DataValuesRepository";
+import { DataSetsRepository } from "domain/repositories/DataSetsRepository";
+import { OrgUnitRepository } from "domain/repositories/OrgUnitRepository";
+import { DataSet } from "domain/entities/DataSet";
+import { OrgUnit } from "domain/entities/OrgUnit";
+import { DataSetExecutionRepository } from "domain/repositories/DataSetExecutionRepository";
+import { promiseMap } from "data/dhis2-utils";
+import { UserRepository } from "domain/repositories/UserRepository";
+import { NotificationsRepository } from "domain/repositories/NotificationsRepository";
+import { DataValue } from "domain/entities/DataValue";
+import { User } from "domain/entities/User";
+import { Path } from "domain/entities/Base";
+import { Async } from "domain/entities/Async";
+import { Settings, SettingsRepository } from "domain/repositories/SettingsRepository";
+
+type EmailTemplate = {
+    subject: string;
+    body: string;
+};
+
+export type NotificationDetail = {
+    lastDateSentEmail: string;
+    lastUpdated: string;
+};
+
+export type NotificationStore = Record<string, NotificationDetail>;
+
+type OptionsUseCase = {
+    storage: string;
+    jsonSettingsPath: string;
+    jsonExecutionsPath: string;
+    dataStoreNameSpace: string;
```

Comment:
[clean arch] This is a bit complex, but the idea is: if we have chosen to hide the implementation (dataStore) behind a repo, no reference to the dataStore should appear in the domain (a use case, here). We have some generic "path" that's being interpreted in each repository (as a namespace/key, as a file path, whatever).